WordPress/gutenberg
# Details Block: Rmove unnecessary comment attributes

## What?
This PR changes the attributes definition of the summary property so that content is not stored in the comment delimiter.

![code-editor](https://github.com/WordPress/gutenberg/assets/54422211/2dc688c5-ca92-4601-8db2-2d68277325da)

## Why?
My understanding is that the values of attributes should be extracted from the markup as much as possible to reduce the amount of content stored in the database.

## How?
~~Add a deprecation.~~ Add `selector` and `source` to `summary` attribute and updated the existing fixture.

## Testing Instructions
- In the trunk branch, insert a Details block and enter summary text and content.
- When you switch to the code editor, make sure it is saved as a comment, like `{"summary":"Details Summary"}`.
- Switch to this PR branch.
- ~~When the editor is reloaded, confirm that the browser console outputs a log indicating that a block deprecation has been performed.~~ Update the contents of the block.
- Confirm that the comment delimiter for the summary property has disappeared from the HTML markup.
- Then, update and save the contents of the block, reload the browser, and confirm that the block is not broken.
